macosx/QuickLookExtension/PreviewProvider.mm

```diff
@@ -0,0 +1,290 @@
+// This file Copyright © Transmission authors and contributors.
+// It may be used under the MIT (SPDX: MIT) license.
+// License text can be found in the licenses/ folder.
+
+#import "PreviewProvider.h"
+
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include <libtransmission/torrent-metainfo.h>
+#include <libtransmission/utils.h>
+
+#import "NSStringAdditions.h"
+
+static NSUInteger const kIconWidth = 16;
+
+namespace
+{
+
+class FileTreeNode
+{
+  public:
+    FileTreeNode() = default;
+    ~FileTreeNode() = default;
+
+    auto MaybeCreateChild(std::string_view child_name)
+    {
+        return children_.try_emplace(std::string{ child_name });
+    }
+
+  private:
+    FileTreeNode(FileTreeNode const&) = delete;
+    FileTreeNode& operator=(FileTreeNode&) = delete;
+
+    std::unordered_map<std::string, FileTreeNode> children_;
+};
+
+} // namespace
+
+NSString* generateIconData(UTType* type, NSUInteger width, NSMutableDictionary<NSString*, QLPreviewReplyAttachment*>* allImgProps)
+{
+    // We need to do this once per file type, per image size
+    NSString* iconFileName = [NSString stringWithFormat:@"%@.%ld.tiff", type.identifier, width];
+
+    if (!allImgProps[iconFileName])
+    {
+        NSImage* icon = [[NSWorkspace sharedWorkspace] iconForContentType:type];
```

Comment:
As `sharedWorkspace` is a property, the general convention in the project is to use property accessors

Reply:
Done in 37b22630c1217893d5e0da6bab7272f7acfcbc6e